src/main/java/org/candlepin/async/tasks/CloudAccountOrgSetupJob.java

```diff
@@ -0,0 +1,227 @@
+            }
+
+            this.setJobArgument(CLOUD_ACCOUNT_ID, cloudAccountId);
+
+            return this;
+        }
+
+        /**
+         * Sets the cloud offering ID for this job
+         *
+         * @param cloudOfferingId
+         *     the cloud offering ID to set for this job
+         * @return a reference to this job config
+         */
+        public CloudAccountOrgSetupJobConfig setCloudOfferingId(String cloudOfferingId) {
+            if (cloudOfferingId == null || cloudOfferingId.isEmpty()) {
+                throw new IllegalArgumentException("cloudOfferingId is null or empty");
+            }
+
+            this.setJobArgument(OFFERING_ID, cloudOfferingId);
+
+            return this;
+        }
+
+        /**
+         * Sets the cloud provider for this job
+         *
+         * @param cloudProviderShortName
+         *     shortcut of the cloud provider
+         * @return a reference to this job config
+         */
+        public CloudAccountOrgSetupJobConfig setCloudProvider(CloudProvider cloudProviderShortName) {
+            if (cloudProviderShortName == null) {
+                throw new IllegalArgumentException("cloudProviderShortName is null");
+            }
+
+            this.setJobArgument(CLOUD_PROVIDER, cloudProviderShortName);
+
+            return this;
+        }
+
+        /**
+         * Sets the owner key for this job
+         *
+         * @param ownerKey
+         *     the owner key to set for this job
+         * @return a reference to this job config
+         */
+        public CloudAccountOrgSetupJobConfig setOwnerKey(String ownerKey) {
+            if (ownerKey.isEmpty()) {
+                throw new IllegalArgumentException("ownerKey is empty");
+            }
+
+            this.setJobArgument(OWNER_KEY, ownerKey);
+
+            return this;
+        }
+
+        @Override
+        public void validate() throws JobConfigValidationException {
+            super.validate();
+
+            try {
+                JobArguments arguments = this.getJobArguments();
+
+                String accountId = arguments.getAsString(CLOUD_ACCOUNT_ID);
+                String offeringId = arguments.getAsString(OFFERING_ID);
+                String cloudProviderShortName = arguments.getAsString(CLOUD_PROVIDER);
+
+                if (accountId == null || accountId.isEmpty()) {
+                    String errmsg = "Cloud Account ID has not been set, or is empty";
+                    throw new JobConfigValidationException(errmsg);
+                }
+
+                if (offeringId == null || offeringId.isEmpty()) {
+                    String errmsg = "Cloud Offering ID has not been set, or is empty";
+                    throw new JobConfigValidationException(errmsg);
+                }
+
+                if (cloudProviderShortName == null) {
+                    String errmsg = "Cloud provider has not been set";
+                    throw new JobConfigValidationException(errmsg);
+                }
```

Comment:
Let's add a check for the ownerKey being blank here. null is allowed so we don't throw an exception in that case, but an empty/blank string should be an error

Reply:
done